### ISSUE TITLE
High-contrast theme should be default when OS is in High-Contrast Mode

### ISSUE THREAD

**Issue by bgashler1** [Microsoft/vscode#4385]
If a user's OS is in high-contrast mode, then it's important we can default him/her to the high-contrast theme.  Depending on visual acuity, switching to it could be difficult if she/he is not already in it.

For example, Chromium detects if Windows is in high-contrast mode (maybe we can learn from their way of doing it).  Perhaps, this is partly something we need Electron to adopt.

**Comment by geirsagberg** [comment]
This is already happening on Windows. However, it should be possible to disable IMHO; I am using a High Contrast theme in Windows, but prefer the Dark+ theme in VSCode. As it is now, every time VSCode starts or I open a new folder, the theme resets to High Contrast and I have to manually switch it back to Dark+.

Please add a setting like `window.autoHighContrastTheme` that may be enabled as default but can be disabled for use cases like mine.

**Comment by geirsagberg** [comment]
The relevant code lines where a setting check would have to be implemented:
https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/electron-browser/integration.ts#L157-L168
or
https://github.com/Microsoft/vscode/blob/master/src/vs/code/electron-main/window.ts#L373-L382

**Comment by bpasero** [comment]
Closing, we implemented this for windows.
